### PR TITLE
Tx serialisation for transfer type

### PR DESCRIPTION
**Pull request by kilic**

### CODE REVIEW

File: contracts/libs/Types.sol [hunk 1]
```diff
@@ -37,6 +37,16 @@ library Types {
...
+        uint256 txType;
+        uint256 fromIndex;
+        uint256 toIndex;
+        uint256 tokenType;
+        uint256 nonce;
+        uint256 amount;
+        bytes signature;
+    }
+
     struct Transaction {
```
[the remark this answers is not in the document]
Comment (kilic) [reply contracts/libs/Types.sol:50]: Yes `Transaction` name or struct used in many places.
Maybe better change it gradually

File: contracts/libs/RollupUtils.sol [hunk 2]
```diff
@@ -394,10 +394,24 @@ library RollupUtils {
...
+        uint256 tokenType,
+        uint256 nonce,
+        uint256 amount,
+        bytes memory signature
+    ) public pure returns (bytes memory) {
+        return
+            abi.encode(txType, from, to, tokenType, nonce, amount, signature);
+    }
+
     function BytesFromTxDeconstructed(
```
[the remark this answers is not in the document]
Comment (kilic) [reply contracts/libs/RollupUtils.sol:415]: the one without signature is used to construct the tx msg that's why we needed another one

File: contracts/libs/Types.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply contracts/libs/Types.sol:50]: Sounds good.

File: contracts/libs/RollupUtils.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (vaibhavchellani) [reply contracts/libs/RollupUtils.sol:415]: Yeah Ill just pass empty from the node.

**Comment by vaibhavchellani** [comment]
Great work on the PR! Can you point me to the functions for compression and decompression?

**Pull request merged**